### PR TITLE
spike: foundation for nextjs-publishing

### PR DESCRIPTION
**Pull request by kevinslin**

### CODE REVIEW

**Review by namjul — COMMENTED** [review]
Thoughts I have in this:
- Would be great to have a how to test description
  * Tried to run in locally but stopped where `data/notes.json` could not be found
- I assume the styles sections was copied from `dendron-next-server`. We slowly start to accumulate more tech dept on the styling side and I am a bit worried how we are going to "cleanup".
- the global eslint rules are currently not enabled for this new packages, equally for the rest apart from `plugin-core`

**Comment by namjul** [comment]
Wondering how/why this PR got added to master in https://github.com/dendronhq/dendron/commit/16b0e5c59e3ee11530199b5c9a11a58f05e14a93

File: .eslintrc.js [hunk 1]
```diff
@@ -77,11 +77,14 @@ module.exports = {
...
-    "react/no-children-prop": "off",
-    "react/prop-types": "off",
     "import/order": "off",
     "import/no-cycle": "off",
+    // --- Reac
+    "react/no-children-prop": "off",
+    "react/prop-types": "off",
+    // suppress errors for missing 'import React' in files
+    "react/react-in-jsx-scope": "off",
+    "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx", ".ts", ".tsx"] }],
```
[the remark this answers is not in the document]
Comment (kevinslin) [reply .eslintrc.js:87]: good catch - removed

File: .eslintrc.js [hunk 2]
```diff
@@ -77,11 +77,14 @@ module.exports = {
 
     "no-shadow": "off",
     "@typescript-eslint/no-shadow": "off", // TODO should be turned on
-
-    "react/no-children-prop": "off",
-    "react/prop-types": "off",
     "import/order": "off",
     "import/no-cycle": "off",
+    // --- Reac
+    "react/no-children-prop": "off",
```
[the remark this answers is not in the document]
Comment (kevinslin) [reply .eslintrc.js:83]: we don't. just updated

File: .eslintrc.js [hunk 3]
```diff
@@ -77,11 +77,14 @@ module.exports = {
...
+    "react/prop-types": "off",
+    // suppress errors for missing 'import React' in files
+    "react/react-in-jsx-scope": "off",
```
[the remark this answers is not in the document]
Comment (kevinslin) [reply .eslintrc.js:86]: nextjs auto injects React

File: .eslintrc.js [hunk 4]
```diff
@@ -77,11 +77,14 @@ module.exports = {
```
[the remark this answers is not in the document]
Comment (kevinslin) [reply .eslintrc.js:77]: we originally had too many issues with eslint. the plan is to introduce this again to all packages . meanwhile enabled it for nextjs-template

**Comment by kevinslin** [comment]
Merged this into master for ease of development. PR updates here -> https://github.com/dendronhq/dendron/pull/1114

**Pull request closed without merging**